### PR TITLE
added swagger ui and configured it

### LINKED ISSUE #9: Setup swagger ui for api documentation
setup swaggerui/openapi for api documentation, so that frontend developers will be able to know each an every api contract easily

references: https://docs.nestjs.com/openapi/introduction

### PR DESCRIPTION
**Pull request by NeelParihar**
**Description**

Added swagger ui and configured, customized it according to the project.

**Related Issue** 
- Fixes #9 

**Type of Change:**

- [x] Code
- [x] User Interface
- [x] New Feature
- [x] Documentation
- [x] Testing

**Checklist**

- [x] My code follows the code style of this project.
- [x] My UI is responsive
- [x] My changes generate no new warnings
- [x] The title of my pull request is a short description of the requested changes.

**Screenshots**
Screenshot 2021-03-17 at 6 01 07 PM

### CODE REVIEW

**Comment by Abhishek-kumar09** [comment]
rebase

File: src/main.ts [hunk 1]
```diff
@@ -1,10 +1,46 @@
...
+      'https://codeforcause.org/',
+      'team@codeforcause.org',
+    )
+    .setLicense(
+      'MIT',
+      'https://github.com/codeforcauseorg/edu-server/blob/master/LICENSE',
+    )
+    .addServer('http://localhost:5000/', 'Development Server')
+    .addBearerAuth()
+    .build();
```
Comment (Abhishek-kumar09) [inline src/main.ts:28]: Move swagger config to another file and import it here to be used

File: src/main.ts [hunk 2]
```diff
@@ -1,10 +1,46 @@
...
+
+  // adding custom options
+  const customOptions: SwaggerCustomOptions = {
+    swaggerOptions: {
+      persistAuthorization: true,
+    },
+    customCss: '.swagger-ui .topbar { display: none }',
+    customSiteTitle: 'Edu Server API Docs',
+    customfavIcon: 'https://codeforcause.org/favicon.ico',
+  };
```
Comment (Abhishek-kumar09) [inline src/main.ts:38]: Add custom options in that file too, and export these constants

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]

File: src/main.ts (hunk 2 above)
Comment (NeelParihar) [reply src/main.ts:38]: should i create a docs folder in the root of the project and add the swagger file in that folder? @Abhishek-kumar09

Comment (Abhishek-kumar09) [reply src/main.ts:38]: create a swagger folder in src and create a new file to put these.

**Comment by NeelParihar** [comment]
please have a look @Abhishek-kumar09

**Comment by Abhishek-kumar09** [comment]
Github, rebase

**Comment by Abhishek-kumar09** [comment]
rebase

**Review by Abhishek-kumar09 — COMMENTED** [review]
Changes looks good, also add this thing in documentation(readme) that how one can go and see docs through swagger.

Mention that the swagger is available on `/api/docs` routes.

**Comment by NeelParihar** [comment]
Oh, yes that make sense will do that

**Comment by Abhishek-kumar09** [comment]
Thanks @NeelParihar

**Pull request merged**